packages/service-library/src/data-providers/website-scan-result-provider.ts

```diff
@@ -54,16 +74,52 @@ export class WebsiteScanResultProvider {
         )) as WebsiteScanResult;
     }
 
+    public normalizeToDbDocument(websiteScanResult: Partial<WebsiteScanResult>): WebsiteScanResult {
+        const documentId = this.getWebsiteScanId(websiteScanResult);
+        const partitionKey = websiteScanResult.partitionKey ?? this.getPartitionKey(documentId);
+
+        return {
+            ...(websiteScanResult as WebsiteScanResult),
+            id: documentId,
+            partitionKey: partitionKey,
+            itemType: ItemType.websiteScanResult,
+        };
+    }
+
     private async mergeAndWrite(
         storageDocument: WebsiteScanResult,
         websiteScanResult: Partial<WebsiteScanResult>,
     ): Promise<WebsiteScanResult> {
-        const mergedDocument = _.mergeWith(storageDocument, websiteScanResult, (target, source, key) => {
+        const mergedDocument = this.merge(websiteScanResult, storageDocument);
+
+        return (await this.cosmosContainerClient.writeDocument<WebsiteScanResult>(mergedDocument)).item;
+    }
+
+    private mergeBatch(sourceDocuments: WebsiteScanResult[]): WebsiteScanResult {
+        if (sourceDocuments.length === 1) {
+            return sourceDocuments[0];
+        }
+
+        let targetDocument: WebsiteScanResult = sourceDocuments[0];
+        sourceDocuments.map((sourceDocument) => {
+            targetDocument = this.merge(sourceDocument, targetDocument);
```

Comment:
Will we need to worry about the possibility of merging several documents with different etags?

Reply:
No, `mergeOrCreate()` function will handle etag logic of the final document before writing into storage